pkg/apis/autoscaling/annotation_validation.go

```diff
@@ -161,6 +161,9 @@ func validateWindow(annotations map[string]string) *apis.FieldError {
 		if annotations[ClassAnnotationKey] == HPA && annotations[MetricAnnotationKey] == CPU {
 			return apis.ErrInvalidKeyName(WindowAnnotationKey, apis.CurrentField, fmt.Sprintf("%s for %s %s", HPA, MetricAnnotationKey, CPU))
 		}
+		if annotations[ClassAnnotationKey] == HPA && annotations[MetricAnnotationKey] == Memory {
+			return apis.ErrInvalidKeyName(WindowAnnotationKey, apis.CurrentField, fmt.Sprintf("%s for %s %s", HPA, MetricAnnotationKey, Memory))
+		}
```

Comment:
Should we fold this into one check with the above? Tbh, I don't even know what we're checking here? If anything, we might want to say that you cannot use window in conjunction with `hpa` :thinking: